### ISSUE TITLE
terser mangled my optional chaining operator

### ISSUE THREAD

**Issue by isaacl** [terser/terser#1040]
**tl;dr** terser replaced `lastModifiedDate?.format(MB_DATE_FORMAT)` with `(void 0)(MB_DATE_FORMAT)`.

## Details

I don't know exactly what terser cmd to repro.  This code was minimized using [terser-webpack-plugin](https://github.com/webpack-contrib/terser-webpack-plugin)....

**No terser:**
```js
const getClassAndVisits = (siteId, classId, lastModifiedDate) => axios_mbAgentRe.get(siteId, 'class/classvisits', {
    params: {
        classId,
        LastModifiedDate: lastModifiedDate?.format(MB_DATE_FORMAT),
    },
})
    .then(({ data }) => data.Class);
```

**With terser:**
```js
(e,t,r)=>x(e,"class/classvisits",{params:{classId:t,LastModifiedDate:(void 0)(je)}
```

I suspect terser realized that my code never called this function with three arguments, so lastModifiedDate was always undefined.  However, it minimized the elvis operator incorrectly.  It should have resolved to `LastModifiedDate:void 0`.

## Other relevant information:
terser: 5.7.1
terser-webpack-plugin: 5.1.4
webpack version:  5.47.1
node.js version: v16.5.0

I was [told](https://github.com/webpack/webpack/issues/13895) to post here...

**Comment by isaacl** [comment]
I'm not sure if it's clear from the spec, maybe I'm better using two optional chaining operators: `obj?.format?.(VAL)`.

But the code before minimization DOES work in node... So I think it's a terser bug?

**Comment by dnalborczyk** [comment]
you can try terser here: https://try.terser.org/

```js
export const getClassAndVisits = (siteId, classId, lastModifiedDate) => axios_mbAgentRe.get(siteId, 'class/classvisits', {
    params: {
        classId,
        LastModifiedDate: lastModifiedDate?.format(MB_DATE_FORMAT),
    },
})
.then(({ data }) => data.Class);
```

```js
export const getClassAndVisits=(s,a,t)=>axios_mbAgentRe.get(s,"class/classvisits",{params:{classId:a,LastModifiedDate:t?.format(MB_DATE_FORMAT)}}).then((({data:s})=>s.Class));
```

looks like it's working correct.

@isaacl I would check the webpack/webpack terser plugin version you are using and which terser version is referenced. possibly it's outdated.

**Comment by isaacl** [comment]
Successfully reproduced with:

```json
{
  module: true,
  compress: {
    passes: 2
  },
  mangle: {},
  output: {},
  parse: {},
  rename: {},
}
```

and 

```js
// write or paste code here

const getClassAndVisits = (siteId, classId, lastModifiedDate) => axios_mbAgentRe.get(siteId, 'class/classvisits', {
    params: {
        classId,
        LastModifiedDate: lastModifiedDate?.format(MB_DATE_FORMAT),
    },
})
    .then(({ data }) => data.Class);


export const getClasses2 = getClassAndVisits(1, 2);
```

**Comment by isaacl** [comment]
bueler?

**Comment by isaacl** [comment]
Here's a minimal example that still breaks:

```js
var t; export const foo= t?.format();
```

default options convert to 
```js
export const foo=(void 0)();
```

**Comment by fabiosantoscode** [comment]
Hey! Thanks for the minimal repro.

This made me feel pretty dumb -- there's actually a test that makes sure the behaviour you're seeing is happening!

I just fixed this but don't want to push because I'm not sufficiently confident in the fix.
